src/viper/trans.ml

```diff
@@ -859,3 +938,44 @@ and tuple_alloc at ctxt lhs ts es : stmt list =
 and tuple_prj ctxt at e1 e2 t =
   prjE at (exp ctxt e1) (exp ctxt e2) (typed_field t)
 
+and label_expr_alloc ~label_id ~label_type ~label_rhs ~label_note at ctxt lhs : seqn' =
+  (* Here is a hack: instead of assigning a unit expression
+     we'll just treat them as statements. The next Motoko's code
+
+     {[
+      let x = label unit_lbl : () { break unit_lbl(v := 42); unreachable_expr; };
+     ]}
+
+     would be translated to
+
+     {[
+      v := 42;
+      goto unit_lbl;
+      unreachable_expr;
+      label unit_lbl;
+     ]} *)
```

Comment:
Yes, but this comment mostly is about last line. While whole code after is about adding explicit `break` to the last expression which is returning from the block.
Add comment about this as well, maybe it would be nicer if you show example of Motoko to Motoko translation.

Reply:
Moved this comment and added an extra one